src/import_edf.jl

```diff
@@ -189,12 +189,30 @@ function edf_signals_to_samplesinfo(edf::EDF.File, edf_signals::Vector{<:EDF.Sig
     return info
 end
 
+struct SamplesInfoError <: Exception
+    msg::String
+    cause::Exception
+end 
+
+function groupby(f, list)
+    d = Dict()
+    foreach(list) do v
+        push!(get!(d, f(v), []), v)
```

Comment:
I think this will introduce type instability.
```suggestion
        push!(get!(d, f(v), Vector{eltype(list)}()), v)
```